tasks/backfill_gh_app_installations.py

```diff
@@ -0,0 +1,149 @@
+import logging
+from typing import List, Optional
+
+import shared.torngit as torngit
+from sqlalchemy import String
+from sqlalchemy.orm.session import Session
+
+from app import celery_app
+from celery_config import backfill_gh_app_installations
+from database.models.core import GithubAppInstallation, Owner, Repository
+from services.owner import get_owner_provider_service
+from tasks.base import BaseCodecovTask
+
+log = logging.getLogger(__name__)
+
+
+class BackfillGHAppInstallationsTask(
+    BaseCodecovTask, name=backfill_gh_app_installations
+):
+    # Looping and adding all repositories in the installation app
+    async def add_repos_service_ids_from_provider(
+        self,
+        db_session: Session,
+        ownerid: int,
+        owner_service: torngit.base.TorngitBaseAdapter,
+        gh_app_installation: GithubAppInstallation,
+    ):
+        repos = await owner_service.list_repos_using_installation()
+        if repos:
+            # Fetching all repos service ids we have for that owner in the DB
+            repo_service_ids_in_db = [
+                repo.service_id
```

Comment:
Unsure on the syntax here, will this be n+1 or does it roll it up?

Reply:
Yeah this should be 1 query, this is the same as

```
repo_list = db_session.query(Repository.service_id).filter_by(ownerid=ownerid).all() // 1 query

for repo in repo_list:
  repo_service_ids_in_db.append(repo.service_id)
```